lib/saml_idp/service_provider.rb

```diff
@@ -19,6 +19,10 @@ def valid?
       attributes.present?
     end
 
+    def valid_identifier?
+      valid? && !identifier.nil?
```

Comment:
Agreed.  I think `valid?` is where the bug is.  Its returning true things that are not valid